frontend/src/components/cluster/ClusterGroupErrorMessage.tsx

```diff
@@ -1,63 +1,66 @@
-import { InlineIcon } from '@iconify/react';
-import { Box, Typography, useTheme } from '@mui/material';
+import { Alert, AlertTitle, Box, Button } from '@mui/material';
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { ApiError } from '../../lib/k8s/apiProxy';
+import { useClusterGroup } from '../../lib/k8s';
+import { ApiError } from '../../lib/k8s/api/v2/ApiError';
 
 export interface ClusterGroupErrorMessageProps {
   /**
-   * A message to display when clusters fail to load resources.
-   * This is used if it is passed in, otherwise a message made from clusterErrors is used.
+   * Array of errors
    */
-  message?: string;
-  /**
-   * Either an array of errors, or keyed by cluster name, valued by the error for that cluster.
-   */
-  clusterErrors?: { [cluster: string]: string | ApiError | null };
+  errors?: ApiError[] | null;
 }
 
-/**
- * filter out null errors
- * @returns errors, but without any that have null values.
- */
-function cleanNullErrors(errors: ClusterGroupErrorMessageProps['clusterErrors']) {
-  if (!errors) {
-    return {};
+export function ClusterGroupErrorMessage({ errors }: ClusterGroupErrorMessageProps) {
+  if (!errors || errors?.length === 0) {
+    return null;
   }
-  const cleanedErrors: ClusterGroupErrorMessageProps['clusterErrors'] = {};
-  Object.entries(errors).forEach(([cluster, error]) => {
-    if (error !== null) {
-      cleanedErrors[cluster] = error;
-    }
-  });
-
-  return cleanedErrors;
+
+  return errors.map((error, i) => <ErrorMessage error={error} key={error.stack ?? i} />);
 }
 
-export function ClusterGroupErrorMessage({
-  clusterErrors,
-  message,
-}: ClusterGroupErrorMessageProps) {
+function ErrorMessage({ error }: { error: ApiError }) {
   const { t } = useTranslation();
-  const theme = useTheme();
-  const clusterObj = cleanNullErrors(typeof clusterErrors === 'object' ? clusterErrors : {});
+  const showClusterName = useClusterGroup().length > 1;
+  const [showMessage, setShowMessage] = useState(false);
 
-  if ((!clusterErrors && !message) || Object.keys(clusterObj).length === 0) {
-    return null;
+  const defaultTitle = t('Failed to load resources');
+  const forbiddenTitle = t("You don't have permission to view this resource");
```

Comment:
(Should be "permission**s**")
If we used the resource kind here, would it make it more clear in the Roles and ClusterRoles case?
`You don't have permissions to view {{ resource.kind }} resources`
I know in this case we have to make sure we distinguish between a list view or a details view, but maybe that's a good thing (besides solving the weird case of repeated error messages because its referring to ClusterRoles and Roles)?

Reply:
I thought about it a bit and this message is vague on purpose, there's more information about the error in the details. Which shows actually which permissions are missing, what role, etc. 

I think this component is generic enough to use for all kinds of places, not just for listing resources